python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -123,6 +140,58 @@ def __init__(cls, name, bases, dict):
                 _METRICS.append(metric)
 
 
+def estimate_object_store_memory_per_node(
+    op: PhysicalOperator, state: "OpState"
+) -> Dict[str, int]:
+    usage_per_node = defaultdict(int)
+
+    # iterate through inqueue
+    for bundle in op._metrics._internal_inqueue._bundle_to_nodes:
+        for _, meta in bundle.blocks:
+            node_id = (
+                meta.exec_stats.node_id
+                if meta.exec_stats and meta.exec_stats.node_id
+                else _NODE_UNKNOWN
+            )
+            usage_per_node[node_id] += meta.size_bytes
+
+    # iterate through outqueue
+    for bundle in op._metrics._internal_outqueue._bundle_to_nodes:
+        for _, meta in bundle.blocks:
+            node_id = (
+                meta.exec_stats.node_id
+                if meta.exec_stats and meta.exec_stats.node_id
+                else _NODE_UNKNOWN
+            )
+            usage_per_node[node_id] += meta.size_bytes
```

Comment:
This will be double-counting blocks (in in-queue and out-queue)

Reply:
@alexeykudinkin could you help me understand how this double counts?